lib/order.rb

```diff
@@ -0,0 +1,81 @@
+require 'csv' #ruby reads the csv file
+require_relative 'customer' #knows about customer class
+
+#assume there is only one of each product
+class Order
+
+  attr_reader :id
+  attr_accessor :products,  :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id #number
+    @products = products #{"melon" => 9.99, "chips" => 3.00} ] cost from cvs?
+    @customer = customer #from customer class
+    @fulfillment_status = fulfillment_status
+
+    raise ArgumentError.new("Invalid fulfillment status.") if [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status) != true
```

Comment:
Nice post-fix conditional! :)

EXTREMELY minor refactoring suggestion, you COULD do this

```suggestion
    raise ArgumentError.new("Invalid fulfillment status.") if ![:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status)
```